### DISCUSSION TITLE
Auto Load External Audio Track if there's no audio track in MP4 file.

### DISCUSSION THREAD

**Discussion by chiaweilee** [iina/iina#5315] [Ideas]
![Screenshot 2024-12-20 at 21 33 29](https://github.com/user-attachments/assets/6dc31e79-aa2f-479b-880d-a33901872f66)

Searching by the filename of mp4, It auto load m4a file as the external Audio Track.

**Comment by low-batt** [comment]
For audio/video and subtitle rendering IINA is using a library from the [mpv project](https://github.com/mpv-player/mpv).  Many mpv features not exposed in IINA's UI can still be used with IINA.

From the `mpv` manual entry for the [audio-file-auto](https://mpv.io/manual/stable/#options-no-audio-file-auto) option:
> **--audio-file-auto=<no|exact|fuzzy|all>, --no-audio-file-auto**
>Load additional audio files matching the video filename. The parameter specifies how external audio files are matched.
>
>  -  **no:**	Don't automatically load external audio files (default).
>  -  **exact:**	Load the media filename with audio file extension.
>   -  **fuzzy:**	Load all audio files containing the media filename.
> -   **all:**	Load all audio files in the current and --audio-file-paths directories.

To enable this feature follow these instructions:
- Start IINA
- Click on  `Settings…` under the `IINA` menu
- The settings panel appears
- On the left side of the panel click on `Advanced`
- Slide the `Enable advanced settings` toggle button to be on (blue)
- In the `Additional mpv options` section click on `+`
- A new entry appears in the table
- Double click on `name` and replace it with `audio-file-auto`
- Double click on `value` and replace it with `fuzzy`
- **Restart IINA** to activate the new settings

Direct support for this option in IINA has been requested in issue #4794.
